packages/aws-cdk-lib/aws-lambda-destinations/lib/s3.ts

```diff
@@ -0,0 +1,24 @@
+import { Construct } from 'constructs';
+import * as lambda from '../../aws-lambda';
+import * as s3 from '../../aws-s3';
+
+/**
+ * Use a S3 bucket as a Lambda destination
+ */
+export class S3Destination implements lambda.IDestination {
+  constructor(private readonly bucket: s3.IBucket) {
+  }
+
+  /**
+   * Returns a destination configuration
+   */
+  public bind(_scope: Construct, fn: lambda.IFunction, _options?: lambda.DestinationOptions): lambda.DestinationConfig {
+    // grant read and putObject permissions
+    this.bucket.grantRead(fn);
```

Comment:
I check other destinations and they only need `Write` permissions. Why do we need `read` permission here?